src/python/pants/backend/python/rules/python_fmt.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import os
+
+from pants.backend.python.rules.create_requirements_pex import (RequirementsPex,
+                                                                RequirementsPexRequest)
+from pants.backend.python.subsystems.black import Black
+from pants.backend.python.subsystems.python_setup import PythonSetup
+from pants.backend.python.subsystems.subprocess_environment import SubprocessEncodingEnvironment
+from pants.engine.fs import Digest, DirectoriesToMerge, PathGlobs, Snapshot
+from pants.engine.isolated_process import ExecuteProcessRequest, ExecuteProcessResult
+from pants.engine.legacy.structs import (PantsPluginAdaptor, PythonAppAdaptor, PythonBinaryAdaptor,
+                                         PythonTargetAdaptor, PythonTestsAdaptor)
+from pants.engine.rules import UnionRule, optionable_rule, rule
+from pants.engine.selectors import Get
+from pants.rules.core.fmt import FmtResult, FmtTarget
+from pants.util.objects import datatype
+
+
+class PythonFormatable(datatype(['target'])):
```

Comment:
Maybe `FormattablePythonTarget`?

Reply:
Also, please add a link to https://github.com/pantsbuild/pants/issues/8343 explaining that this is a horrible hack, and why it's there :)